### PR TITLE
Migrate Iceberg tests to BaseConnectorTest

### PR DESCRIPTION
**Pull request by wendigo**

### CODE REVIEW

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                return super.hasBehavior(connectorBehavior);
+        }
+    }
+
+    @Test
+    @Override
+    public void testRenameTable()
+    {
+        assertThatThrownBy(super::testRenameTable)
+                .hasMessage("Rename not supported for Iceberg tables");
```
Comment (findepi) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java:65]: shouldn't `hasBehavior(SUPPORTS_RENAME_TABLE) = false` be enough?

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    }
+
+    @Test
+    @Override
+    public void testDelete()
+    {
+        assertThatThrownBy(super::testDelete)
+                .hasStackTraceContaining("This connector only supports delete where one or more partitions are deleted entirely");
```
Comment (findepi) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java:73]: is there a positive test for delete with iceberg?

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        }
+
+        if (typeName.startsWith("decimal(")
+                || typeName.equals("time")) {
+            // TODO this should either work or fail cleanly
+            return Optional.empty();
+        }
+
+        if (typeName.equals("timestamp")) {
+            return Optional.of(new DataMappingTestSetup("timestamp(6)", "TIMESTAMP '2020-02-12 15:03:00'", "TIMESTAMP '2199-12-31 23:59:59.999999'"));
```
Comment (findepi) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java:139]: Document why

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        if (typeName.equals("tinyint")
+                || typeName.equals("smallint")
+                || typeName.startsWith("char(")) {
+            return Optional.of(dataMappingTestSetup.asUnsupported());
+        }
+
+        if (typeName.startsWith("decimal(")
+                || typeName.equals("time")) {
+            // TODO this should either work or fail cleanly
+            return Optional.empty();
```
Comment (findepi) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java:135]: we should have an issue & fix this

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        }
+
+        if (typeName.equals("timestamp")) {
+            return Optional.of(new DataMappingTestSetup("timestamp(6)", "TIMESTAMP '2020-02-12 15:03:00'", "TIMESTAMP '2199-12-31 23:59:59.999999'"));
+        }
+
+        if (typeName.equals("timestamp(3) with time zone")) {
+            return Optional.of(new DataMappingTestSetup("timestamp(6) with time zone", "TIMESTAMP '2020-02-12 15:03:00 +01:00'", "TIMESTAMP '9999-12-31 23:59:59.999999 +12:00'"));
```
Comment (findepi) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java:143]: document why

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java (hunk 1 above)
Comment (wendigo) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java:65]: Nope. It's not guarded there by behaviour check

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+        return createIcebergQueryRunner(ImmutableMap.of(), REQUIRED_TPCH_TABLES);
+    }
+
+    @Override
+    protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
+    {
+        switch (connectorBehavior) {
+            case SUPPORTS_COMMENT_ON_COLUMN:
+            case SUPPORTS_RENAME_TABLE:
+            case SUPPORTS_AGGREGATION_PUSHDOWN:
```
Comment (hashhar) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java:50]: Is any test in `BCT` using this flag? Remove if unused.

**Review by hashhar — COMMENTED** [review]
LGTM % findepi's comments.

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java (hunk 2 above)
Comment (findepi) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java]: @wendigo ?

**Comment by wendigo** [comment]
PTAL @hashhar @losipiuk

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java (hunk 1 above)
Comment (losipiuk) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java]: :)

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergConnectorTest.java [hunk 7]
```diff
@@ -30,10 +30,13 @@
 import io.trino.spi.statistics.ColumnStatistics;
 import io.trino.spi.statistics.DoubleRange;
 import io.trino.spi.statistics.TableStatistics;
-import io.trino.testing.AbstractTestIntegrationSmokeTest;
+import io.trino.testing.BaseConnectorTest;
```
Comment (losipiuk) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergConnectorTest.java:33]: can you add a preparatory commit which introduces `AbstractTestIcebergConnectorTest.java` based on `TestIcebergConnectorTest` without yet merging `ATISmoke` into it

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergConnectorTest.java [hunk 8]
```diff
@@ -1253,7 +1274,7 @@ public void testBasicTableStatistics()
                         .row("col", null, null, 0.0, null, "-10.0", "200.0")
                         .row(null, null, null, null, 3.0, null, null)
                         .build();
-        assertEquals(result, expectedStatistics);
+        Assert.assertEquals(result, expectedStatistics);
```
Comment (losipiuk) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergConnectorTest.java:1277]: static import?

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergConnectorTest.java [hunk 9]
```diff
@@ -1764,7 +1778,7 @@ public void testIncorrectIcebergFileSizes()
         dataFile.put("file_size_in_bytes", alteredValue);
 
         // Replace the file through HDFS client. This is required for correct checksums.
-        HdfsEnvironment.HdfsContext context = new HdfsContext(getSession().toConnectorSession());
+        HdfsEnvironment.HdfsContext context = new HdfsEnvironment.HdfsContext(getSession().toConnectorSession());
```
Comment (losipiuk) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergConnectorTest.java:1781]: stat import HdfsContext

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/util/PrimitiveTypeMapBuilder.java [hunk 10]
```diff
@@ -42,7 +43,7 @@ private PrimitiveTypeMapBuilder() {}
     private Map<List<String>, Type> buildTypeMap(List<Type> types, List<String> columnNames)
     {
         for (int i = 0; i < types.size(); i++) {
-            visitType(types.get(i), columnNames.get(i), ImmutableList.of());
+            visitType(types.get(i), AvroSchemaUtil.makeCompatibleName(columnNames.get(i)), ImmutableList.of());
```
Comment (hashhar) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/util/PrimitiveTypeMapBuilder.java:46]: Does `visitRowType` also need this?

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergConnectorTest.java [hunk 11]
```diff
@@ -108,7 +109,31 @@ protected AbstractTestIcebergSmoke(FileFormat format)
...
+            default:
+                return super.hasBehavior(connectorBehavior);
+        }
+    }
+
+    @Test
+    @Override
+    public void testDelete()
+    {
+        // Deletes are covered with testMetadata*Delete test methods
```
Comment (hashhar) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergConnectorTest.java:134]: Maybe the method should be renamed to `testRowLevelDelete`?

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergConnectorTest.java [hunk 12]
```diff
@@ -124,10 +149,8 @@ public void testShowCreateSchema()
                         "\\)");
     }
 
-    @Override
     @Test
-    // This particular method may or may not be @Flaky. It is annotated since the problem is generic.
-    @Flaky(issue = "https://github.com/trinodb/trino/issues/5201", match = "Failed to read footer of file: HdfsInputFile")
```
Comment (hashhar) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergConnectorTest.java:130]: Why removed @Flaky?

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergConnectorTest.java [hunk 13]
```diff
@@ -164,8 +185,8 @@ public void testShowCreateTable()
                         "   comment varchar\n" +
                         ")\n" +
                         "WITH (\n" +
-                        "   format = '" + format.name() + "'\n" +
-                        ")");
+                        "   format = '%s'\n" +
```
Comment (hashhar) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergConnectorTest.java:188]: nit: unrelated

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergConnectorTest.java [hunk 14]
```diff
@@ -1786,4 +1800,54 @@ public void testIncorrectIcebergFileSizes()
...
+            return Optional.of(dataMappingTestSetup.asUnsupported());
+        }
+
+        if (typeName.startsWith("decimal(") || typeName.equals("time")) {
+            // Cases covered by testDecimal method
+            return Optional.empty();
+        }
+
+        if (typeName.equals("timestamp")) {
+            return Optional.of(new DataMappingTestSetup("timestamp(6)", "TIMESTAMP '2020-02-12 15:03:00'", "TIMESTAMP '2199-12-31 23:59:59.999999'"));
```
Comment (hashhar) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergConnectorTest.java:1827]: Please add a comment about why this is required. Same for the `timestamp(3) with time zone` branch.

**Review by hashhar — COMMENTED** [review]
Some comments.

Moving the 2nd commit to the beginning would allow some overrides to be removed for the `testRenameTable`.

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/util/PrimitiveTypeMapBuilder.java (hunk 10 above)
Comment (hashhar) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/util/PrimitiveTypeMapBuilder.java:46]: Do we have a test for ROW types with fancy column names? I remember something was added for ORC and case-sensitivity in https://github.com/trinodb/trino/pull/7350/files for example.

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java (hunk 2 above)
Comment (wendigo) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorTest.java]: what do you mean by "positive" test? Iceberg delete cases are covered by "testMetadata*Delete" methods

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/util/PrimitiveTypeMapBuilder.java (hunk 10 above)
Comment (wendigo) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/util/PrimitiveTypeMapBuilder.java:46]: @hashhar visitRowType doesn't need that.

**Comment by wendigo** [comment]
Superseded by https://github.com/trinodb/trino/pull/8065
